pkg/auth/providers/azure/azure_actions.go

```diff
@@ -100,3 +106,52 @@ func (ap *azureProvider) testAndApply(actionName string, action *types.Action, r
 
 	return ap.tokenMGR.CreateTokenAndSetCookie(user.Name, userPrincipal, groupPrincipals, providerToken, 0, "Token via Azure Configuration", request, userExtraInfo)
 }
+
+// migrateToMicrosoftGraph represents the migration of the registered Azure AD auth provider
+// from the deprecated Azure AD Graph API to the Microsoft Graph API.
+// It modifies the existing auth config value in the database, so that it has a new endpoint to the new API.
+// It is also critical that it deletes all existing users' access tokens that had been stored for accessing
+// the deprecated Azure AD Graph API.
+func (ap *azureProvider) migrateToMicrosoftGraph() error {
+	// Delete all users' access tokens to rectify the flaw in the previous security design of Azure AD Graph API access pattern.
+	// It used the Delegated permission types, which required Rancher to retrieve and store all users' credentials to
+	// the Graph API, rather than credentials of just the service account (the application defined in Azure).
+	defer ap.deleteUserAccessTokens()
+	defer groupCache.Purge()
+
+	c, err := ap.getAzureConfigK8s()
+	if err != nil {
+		return err
+	}
+	if authProviderEnabled(c) {
```

Comment:
So if I am reading this right, you have to be active under the old format to use this migrate endpoint. Is there some other part of functionality to make sure that new users in 2.6.6+ only can use the new graph api?